tests/core/templates/test_oa_hash_map.h

```diff
@@ -0,0 +1,193 @@
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_OA_HASH_MAP_H
+#define TEST_OA_HASH_MAP_H
+
+#include "core/templates/oa_hash_map.h"
+#include "scene/resources/texture.h"
+
+#include "tests/test_macros.h"
+
+namespace TestOAHashMap {
+
+TEST_CASE("[OAHashMap] Insert element") {
+	OAHashMap<int, int> map;
+	map.insert(42, 84);
+	int data = 0;
+	bool lookup_res = map.lookup(42, data);
+	int value = *map.lookup_ptr(42);
+	CHECK(lookup_res);
+	CHECK(value == 84);
+	CHECK(data == 84);
+}
+
+TEST_CASE("[OAHashMap] Set element") {
+	OAHashMap<int, int> map;
+	map.set(42, 84);
+	int data = 0;
+	bool lookup_res = map.lookup(42, data);
+	int value = *map.lookup_ptr(42);
+	CHECK(lookup_res);
+	CHECK(value == 84);
+	CHECK(data == 84);
+}
+
+TEST_CASE("[OAHashMap] Overwrite element") {
+	OAHashMap<int, int> map;
+	map.set(42, 84);
+	map.set(42, 1234);
+	int result = *map.lookup_ptr(42);
+	CHECK(result);
+}
+
+TEST_CASE("[OAHashMap] Remove element") {
+	OAHashMap<int, int> map;
+	map.insert(42, 84);
+	map.remove(42);
+	CHECK(!map.has(42));
+}
+
+TEST_CASE("[OAHashMap] Get Num_Elements") {
+	OAHashMap<int, int> map;
+	map.set(42, 84);
+	map.set(123, 84);
+	map.set(123, 84);
+	map.set(0, 84);
+	map.set(123485, 84);
+
+	CHECK(map.get_num_elements() == 4);
+}
+
+TEST_CASE("[OAHashMap] Iteration") {
+	OAHashMap<int, int> map;
+	map.insert(42, 84);
+	map.insert(123, 12385);
+	map.insert(0, 12934);
+	map.insert(123485, 1238888);
+	map.set(123, 111111);
+
+	Vector<Pair<int, int>> expected;
+	expected.push_back(Pair<int, int>(42, 84));
+	expected.push_back(Pair<int, int>(123, 111111));
+	expected.push_back(Pair<int, int>(0, 12934));
+	expected.push_back(Pair<int, int>(123485, 1238888));
+
+	int idx = 0;
+	for (OAHashMap<int, int>::Iterator it = map.iter(); it.valid; it = map.next_iter(it)) {
+		int64_t result = expected.find(Pair<int, int>(*it.key, *it.value));
+		CHECK(result >= 0);
+		++idx;
```

Comment:
Prefer post-incrementation if you don't rely on pre-incrementation behavior:

```suggestion
		idx++;
```